src/Core/Domain/OrderReturn/Exception/MissingOrderReturnRequiredFieldsException.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\OrderReturn\Exception;
+
+/**
+ * Is thrown when some of the required fields are missing when saving order return
+ */
+class MissingOrderReturnRequiredFieldsException extends OrderReturnException
```

Comment:
Is it somehow different than any {domain}ConstraintException with dedicated codes? Couldnt it be implemented as everywhere else? If some field is missing, then OrderReturnConstraintException::{SOME_SPECIFIC_FIELD_IS_INVALID} should be thrown, no?

Reply:
not sure if Im missing something, but if we use ObjectModelValidator and manually validate each field inside, then we have an error code for each field and its never a list of missing fields. (like there is never a list of missing fields in product - instead we have one error which represents one invalid field - either its missing or invalid in any other way)

Reply:
Yeah I don't think this exception is used anymore so I will delete it